CONTRIBUTING.md

```diff
@@ -0,0 +1,303 @@
+`cargo test`.
+
+Tests expected to fail are often embedded directly into rust source code,
+because you usually want to additionally check that the error is the one you
+expect.
+
+## Benchmarking
+
+If your change is likely to impact performance, it is recommended to run the
+benchmark suite on master and on your branch to assess any performance changes.
+Please report your findings in the description of the PR.
+
+The benchmark suite is located in the `benches/` directory. To run it:
+
+```
+cargo bench
+```
+
+Note that a full run takes some time, up to a dozen of minutes. You can run
+specific benchmarks instead of the full suite. Please refer to the documentation
+of [`cargo bench`][doc-cargo-bench].
+
+# How to submit changes
+
+Once you have made sure the maintainers are aware of your changes, you can start
+the implementation, eventually submitting changes as a pull request.
+
+**Try to keep pull requests small and focused**. Avoid packing refactoring,
+cosmetic changes or anything not directly related to your original goal in the
+same pull request. If preliminary steps make sense as standalone changes, don't
+hesitate to split your pull request into several ones.
+
+1. Create a new topic branch: `git checkout -b feature/adding-some-stuff`
+2. Implement your changes and commit them. Try to keep commits focused as well.
+3. Documentation: If you added new items (functions, modules) to the public API
+   of a crate, please document those items in-code. If you made an user-facing
+   change (syntax, stdlib, language features, etc.), please update the existing
+   documentation (in particular the user-manual) in consequence.
+4. Tests: be it for bug fixing or adding new features, try to write extensive
+   tests as much as possible, to ensure the correctness of your changes as well
+   as avoiding regressions.
+5. Make sure your PR will pass the CI:
+    * run `cargo test` on the root crate `nickel-lang`
+    * run `cargo check` inside auxiliary crates: `lsp/nls` and
+        `utilities/`, to make sure that your change didn't impact them
+    * run `cargo fmt` at the root of each modified crate
+    * (optional) check that the WebAssembly REPL is building. See [building the
+      WebAssembly REPL](#webassembly-repl). This step may take some time, and
+      should only be required if you added, upgraded or downgraded
+      dependencies.
+6. Once all the previous steps are completed, you can push your branch to your
+   fork and make a pull request on GitHub.
+
+Your pull request will need at least one approval from a maintainer to be
+merged. Maintainers will review your pull request and may request changes as
+well as leave comments, that needs to be addressed in order to the process to
+move forward.
+
+Even when a code review is accepting, it may still come with questions or
+comments. Please consider all the review comments and decide to act in
+consequence. Once done, resolve the corresponding conversations.
```

Comment:
```suggestion
```
Instead, just let people know when you review their PRs what you expect of them.